packages/world/src/systems/SpawnSystem.sol

```diff
@@ -117,7 +115,7 @@ contract SpawnSystem is System {
   }
 
   function getValidSpawnY(Vec3 spawnCoordCandidate) public view returns (Vec3 spawnCoord) {
-    for (int32 i = 0; i < CHUNK_SIZE; i++) {
+    for (int32 i = CHUNK_SIZE - 1; i >= 0; i--) {
```

Comment:
ahh nice, iterating from top to bottom cuz likely surface is on top so we can early return